ytypes/validate.go

```diff
@@ -22,9 +22,28 @@ import (
 	"github.com/openconfig/ygot/ygot"
 )
 
+// LeafrefOptions controls the behaviour of validation functions for leaf-ref
+// data types.
+type LeafrefOptions struct {
+	// IgnoreMissingData determines whether leafrefs that target a node
+	// that does not exist should return an error to the calling application. When
+	// set to true, no error is returned.
+	//
+	// This functionality is typically used where a partial set of schema information
+	// is populated, but validation is required - for example, configuration for
+	// a protocol within OpenConfig references an interface, but the schema being
+	// validated does not contain the interface definitions.
+	IgnoreMissingData bool
+}
+
+// IsValidationOption ensures that LeafrefOptions implements the ValidationOption
+// interface.
+func (LeafrefOptions) IsValidationOption() {}
+
 // Validate recursively validates the value of the given data tree struct
-// against the given schema.
-func Validate(schema *yang.Entry, value interface{}) util.Errors {
+// against the given schema. The supplied options (opts) are used to determine
```

Comment:
nit: the description of opts doesn't provide a lot of information in this case - suggest dropping that.

Reply:
SGTM!